### PR TITLE
Honor `#[verify_only]` in move-compiler-v2.  Add relevant tests from v1.  Also add tests of `#[deprecated]`.

### LINKED ISSUE #13696: [Bug][move-compiler-v2] It's likely that V2 includes `#[verify_only]` stuff (including use and friend decls) during compilation
# 🐛 Bug

v2 starts with `move_model::run_model_builder_in_compiler_mode()`, which initializes v1 config with `Flags::model_compilation()`, which has `verify: true`.  That means that `#[verify_only]` code will be included.  Functions and modules can be stripped in CGen, but friend/use declarations can also be `#[verify_only]` so there is a potential problem here.  

We need tests to see if this is happening, and turn it off if so.

### LINKED ISSUE #13737: [Bug][move-compiler-v2] When compiling, we should check that missing spec targets were actually removed
# 🐛 Bug

In some cases, a function or struct may be marked `#[verify_only]` or `#[test_only]`.  A corresponding `spec` clause which refers to that function or struct will not find its target if Compiler V2 is run without verification or testing on.

Testing code already should see this problem, but we haven't seen it in practice.  (A `#[test_only]` or `#[test]` function with a separate `spec` clause.)

PR #13732 is going to disable verification by default when compiling Move code with V2.  This leads to a similar problem with verification code, and there is code in the Aptos stdlibs which has such code (first seen with function `shift_left_for_verification_only` defined in `aptos-move/framework/move-stdlib/sources/bit_vector.move`).  This causes a compile-time error when the `spec` target is resolved.

As a quick workaround, PR #13732 disables the requirement that a spec target be resolved if verification or testing are not enabled.  This allows further work to continue, and the specs can be tested by compiling and/or testing in verification mode.

Eventually we should fix this and generate correct  "unresolved spec target" errors for specs even in the presence of `#[test_only]` and `#[verify_only]` declarations.  This may require tracking (and identifying) the elided names, which is not easy because this is done quite early in compilation.

In the long-run, we might keep such conditional definitions so they can be checked even if they are eventually unused.  Due to the possibility of `#[verify_only]` and `#[test_only]` `friend` and `use` declarations, however, representing all 4 possible dependence and resolution graphs at compile time may be infeasible.  But perhaps we can find a clever solution or a middle ground.

However, this bug exists only to track the particular case of generating correct "unresolved spec target" errors in the !verify and !testing case.

### PR DESCRIPTION
**Pull request by brmataptos**
## Description

Add options and plumbing for compile_verify_code, warn_deprecated, etc. to move-compiler-v2, and test add relevant tests from v1.  Requires disabling verify by default in compiler path in move-model.

Fixes #13696.

Merged in a commit from Wolfgang to fix #13737.

## How Has This Been Tested?

Added relevant tests from V1 under tests/more-v1/.  Also ran `move_pr.sh`, which uncovers issue #13737, which is now also fixed here, thanks to Wolfgang.  Added new tests to illustrate #13737: function_with_spec.move to test `#[verify_only]` and `#[test_only]` with associated `spec` blocks for verify and testing on and off.

## Key Areas to Review

Is plumbing correct?

## Type of Change
- [ ] New feature
- [x] Bug fix
- [ ] Breaking change
- [ ] Performance improvement
- [ ] Refactoring
- [ ] Dependency update
- [ ] Documentation update
- [x] Tests

## Which Components or Systems Does This Change Impact?
- [ ] Validator Node
- [ ] Full Node (API, Indexer, etc.)
- [x] Move/Aptos Virtual Machine
- [ ] Aptos Framework
- [ ] Aptos CLI/SDK
- [ ] Developer Infrastructure
- [ ] Other (specify)

## Checklist
- [x] I have read and followed the [CONTRIBUTING](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I identified and added all stakeholders and component owners affected by this change as reviewers
- [x] I tested both happy and unhappy path of the functionality
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: third_party/move/move-compiler-v2/src/options.rs [hunk 1]
```diff
@@ -56,31 +69,52 @@ pub struct Options {
...
     #[clap(long)]
     pub whole_program: bool,
+
     /// Whether to compile #[test] and #[test_only] code
     #[clap(skip)]
     pub compile_test_code: bool,
+
+    /// Whether to compile #[verify_only] code
+    #[clap(skip)]
+    pub compile_verify_code: bool,
```
Comment (rahxephon89) [inline third_party/move/move-compiler-v2/src/options.rs:117]: Could you point to where this flag is used apart from unit tests?

Comment (brmataptos) [reply third_party/move/move-compiler-v2/src/options.rs:117]: `run_move_prover_v2()`, in `third_party/move/move-prover/src/lib.rs`.

**Review by rahxephon89 — APPROVED** [review]

File: third_party/move/move-compiler-v2/src/options.rs [hunk 2]
```diff
@@ -25,29 +31,36 @@ pub struct Options {
         num_args = 0..
     )]
     pub dependencies: Vec<String>,
+
```
Comment (wrwg) [inline third_party/move/move-compiler-v2/src/options.rs:34]: Has this been done by the auto formatter? Please note that there is no agreement in general what is the better style, it actually pretty much depends on the IDE and font you are using. Its not necessarily better now. I would leave such decisions of formatting at the author and not enforce my private style tastes.

File: third_party/move/move-compiler-v2/src/options.rs [hunk 3]
```diff
@@ -56,31 +69,52 @@ pub struct Options {
...
     /// Dependencies to compile but not treat as a test/docgen/warning/prover target.
     /// Each source_dep should be a path to either (1) a Move file or (2) a directory containing
     /// Move files, all to be compiled (e.g., not the root directory of a package---which contains
     /// Move.toml---but a specific subdirectorysuch as `sources`).
     #[clap(skip)]
     pub sources_deps: Vec<String>,
+
+    #[clap(long = cli::MOVE_COMPILER_WARN_OF_DEPRECATION_USE_FLAG,
+           default_value=bool_to_str(move_compiler_warn_of_deprecation_use_env_var()))]
+    pub warn_deprecated: bool,
```
Comment (wrwg) [inline third_party/move/move-compiler-v2/src/options.rs:93]: Just wondering how this relates to the title of the PR. Should you update the title?

**Review by wrwg — APPROVED** [review]

File: third_party/move/move-compiler-v2/src/options.rs (hunk 2 above)
Comment (brmataptos) [reply third_party/move/move-compiler-v2/src/options.rs:34]: I just can't read it the other way.  Call it an accessibility issue for the partially-blind.

File: third_party/move/move-compiler-v2/src/options.rs (hunk 3 above)
Comment (brmataptos) [reply third_party/move/move-compiler-v2/src/options.rs:93]: Fixed it.  Though it's the final commit message that really matters.

**Pull request merged**